EIPS/eip-4804.md

```diff
@@ -74,57 +74,79 @@ The auto mode is the default mode to resolve (also applies when the "resolveMode
 
 Note that if **method** does not exist, i.e., **path** is empty or "/", then the contract will be called with empty calldata.
 
-- **returns** attribute in **query** tells the format of the returned data. If not specified, the returned message data will be parsed in "(bytes32)" and MIME will be set based on the suffix of the last argument. If **returns** is "()", the returned data will be parsed in raw bytes in JSON.  Otherwise, the returned message will be parsed in the specified **returns** attribute in JSON.  If multiple **returns** attributes are present, the value of the last **returns** attribute will be applied. Note that **returnTypes** is the alias of **returns**, but it is not recommended to use and is mainly for backward-compatible purpose.
+- **returns** attribute in **query** tells the format of the returned data. If not specified, the returned message data will be treated in ABI-type "(bytes)" and the decoded bytes will be returned to the frontend. The MIME will be set based on the suffix of the last argument. If **returns** is "()", the returned data will be parsed in raw bytes in JSON format.  Otherwise, the returned message data will be parsed in the specified **returns** attribute in JSON format.  If multiple **returns** attributes are present, the value of the last **returns** attribute will be applied. Note that **returnTypes** is the alias of **returns**, but it is not recommended to use and is mainly for backward-compatible purpose.
```

Comment:
Thanks for the comment.  I wonder whether it may create confusion for the path like "/abc.xyz/13" where we may interpret ".xyz/13" as suffix?  Or I may use "The MIME will be set based on the suffix of the last argument or if there is no argument, the suffix of the method."?

Reply:
Yes ""The MIME will be set based on the suffix of the last argument or if there is no argument, the suffix of the method."" is better indeed!